tests/test_views.py

```diff
@@ -0,0 +1,199 @@
+
+import fixtures
+
+
+class TestDownloadFileView():
+    def setup(self):
+        self._saved__abstract_methods__ = DownloadFileView.__abstractmethods__
+        DownloadFileView.__abstractmethods__ = set()
+
+        self.kwargs = {'arg1': 'foo', 'arg2': 'bar'}
+        self.view = DownloadFileView()
+
+        self._patch_determine_filetype = mock.patch.object(self.view, 'determine_filetype', autospec=True)
+        self._patch_determine_filetype.start()
+        self.view.determine_filetype.return_value = DownloadFileView.FILETYPES.ODS
+
+        self._patch_get_file_context = mock.patch.object(self.view, 'get_file_context', autospec=True)
+        self._patch_get_file_context.start()
+
+        self._patch_create_response = mock.patch.object(self.view, 'create_response', autospec=True)
+        self._patch_create_response.start()
+        self.view.create_response.return_value = (Response(), 200)
+
+        self._patch_init_hook = mock.patch.object(self.view, '_init_hook', autospec=True)
+        self._patch_init_hook.start()
+
+        self._patch_post_request_hook = mock.patch.object(self.view, '_post_request_hook', autospec=True)
+        self._patch_post_request_hook.start()
+
+    def teardown(self):
+        DownloadFileView.__abstractmethods__ = DownloadFileView.__abstractmethods__
+
+        for patch in [self._patch_determine_filetype, self._patch_get_file_context, self._patch_create_response,
+                      self._patch_init_hook, self._patch_post_request_hook]:
+            try:
+                patch.stop()
+            except RuntimeError:
+                pass
+
+    def test_abstract_methods_required_for_instantiation(self):
+        DownloadFileView.__abstractmethods__ = self._saved__abstract_methods__
+
+        with pytest.raises(TypeError) as e:
+            DownloadFileView()
+
+        assert str(e.value) == "Can't instantiate abstract class DownloadFileView with abstract methods " \
+                               "_init_hook, determine_filetype, generate_csv_rows, generate_ods, get_file_context"
+
+    def test_create_blank_ods_with_styles(self):
+        """Assert that, at a minimum, the styles exist."""
+        spreadsheet = self.view.create_blank_ods_with_styles()
+
+        style_names = ['col-default', 'col-wide', 'col-extra-wide',
+                       'row-default', 'row-tall', 'row-tall-optimal',
+                       'cell-default', 'cell-header']
+        for style_name in style_names:
+            assert spreadsheet._document.getStyleByName(six.text_type(style_name)) is not None
+
+    @pytest.mark.parametrize('file_type, call_method',
+                             (
+                                 ('CSV', 'generate_csv_rows'),
+                                 ('ODS', 'generate_ods'),
+                             ))
+    def test_create_response(self, file_type, call_method):
+        kwargs = {'filename': 'test'}
+        self._patch_create_response.stop()
+
```

Comment:
Again I don't think that testing the implementation detail is really the right thing to do here. It leads to a very large amount of test code, anyway. We should be testing what the response looks like (potentially even just comparing to some fixture-blob) rather than worrying about how many times BytesIO had to get called?!

Reply:
I've stripped out some of the more useless bits of the implementation testing, but maybe we can sit together for a bit to go over what else is wrong, as I'm drawing a bit of a blank about where to draw the line for implementation vs result.